src/utilities/activities.ts

```diff
@@ -1,264 +1,9 @@
 import { omitBy } from 'lodash-es';
-import type {
-  ActivitiesMap,
-  Activity,
-  ActivityDirective,
-  ActivityDirectiveId,
-  ActivityUniqueId,
-} from '../types/activity';
+import type { ActivityDirective } from '../types/activity';
 import type { ActivityMetadata, ActivityMetadataKey, ActivityMetadataValue } from '../types/activity-metadata';
-import type { PlanMergeActivityDirectiveSource, PlanMergeActivityDirectiveTarget, PlanSlimmer } from '../types/plan';
-import type { Span, SpanId } from '../types/simulation';
+import type { Span, SpanId, SpansMap } from '../types/simulation';
 import { compare, isEmpty } from './generic';
-import { getDoyTimeFromDuration, getUnixEpochTime } from './time';
-
-type ActivityStartTimesMap = {
-  [activityUniqueId: ActivityUniqueId]: string;
-};
-type TraversalMap = {
-  [activityUniqueId: ActivityUniqueId]: boolean;
-};
-function determineActivityDirectiveStartTime(
```

Comment:
So is the start time now being computed on the backend? The purpose of this was to basically display the directive at the correct time in the timeline based on its anchor.

Reply:
Yeah this was the last piece I have not brought over. I spoke with @Mythicaeda and I think we are going to move this to the back end.

Reply:
Should this not be removed until it's ready on the backend or are we thinking it's not going to be too much time until it's implemented?

Reply:
So it's a little tricky because this PR just uses the `start_offset` to render activity directives, not the DOY. So this function needs to be updated to calculate the correct start_offset in the activity layer itself (since we no longer need to transform the directive when we fetch them, it would not happen in this file). Maybe we can huddle to talk about options?

Reply:
Added back this function and calling it on draw